### PR TITLE
Add remaining pre-commit-hooks

### PR DESCRIPTION
**Pull request by pat-trunk-io**

### CODE REVIEW

**Review by det — APPROVED** [review]

**Comment by det** [comment]
Looks like you need to define the xml file type.

File: linters/pre-commit-hooks/plugin.yaml [hunk 1]
```diff
@@ -19,3 +19,148 @@ lint:
           success_codes: [0, 1]
           files:
             - ALL
+        - name: check-added-large-files
+          enabled: false
+          output: pass_fail
+          read_output_from: stdout
+          run: check-added-large-files ${target}
```
Comment (TylerJang27) [inline linters/pre-commit-hooks/plugin.yaml:26]: Can we add `python3` to the front of these if appropriate? Shebangs don't work for these executions on Windows.

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -115,18 +115,50 @@ jobs:
...
   # Run tests against all linters for known_good_version and latest version
-  linter_tests:
-    name: Linter Tests ${{ matrix.os }}
-    runs-on: [self-hosted, "${{ matrix.os }}"]
+  linter_tests_linux:
+    name: Linter Tests Linux
+    runs-on: public-amd64-2xlarge
+    needs: detect_changes
+    if:
+      needs.detect_changes.outputs.linters == 'true' || needs.detect_changes.outputs.all-linters ==
```
Comment (TylerJang27) [inline .github/workflows/pr.yaml:123]: If we end up keeping this, can we merge it back into the matrix and add a new field to the object for the correct runner name?

File: linters/pre-commit-hooks/plugin.yaml (hunk 1 above)
Comment (pat-trunk-io) [reply linters/pre-commit-hooks/plugin.yaml]: Done

**Review by TylerJang27 — APPROVED** [review]
LGTM

File: linters/pre-commit-hooks/plugin.yaml (hunk 1 above)
Comment (TylerJang27) [reply linters/pre-commit-hooks/plugin.yaml]: False alarm. This was previously WAI, the files are .exe on Windows and inside of `Scripts/`, so no need for the python3 bit.

Comment (pat-trunk-io) [reply linters/pre-commit-hooks/plugin.yaml:26]: Updated

**Pull request merged**